mdw-services/src/com/centurylink/mdw/services/process/ProcessExecutorImpl.java

```diff
@@ -1597,9 +1597,23 @@ private void cancelProcessInstance(ProcessInstance pProcessInst)
                 "ProcessInstance has been cancelled.", null);
         edao.setProcessInstanceStatus(pProcessInst.getId(), WorkStatus.STATUS_CANCELLED);
         edao.removeEventWaitForProcessInstance(pProcessInst.getId());
+        this.cancelErrorHandlers(pProcessInst);
         this.cancelTasksOfProcessInstance(pProcessInst);
     }
 
+    private void cancelErrorHandlers(ProcessInstance procInst) throws Exception {
```

Comment:
- Since processInstance is being passed in, can we call processInstance.getActivities() instead of db retrieval?
 - After finding a candidate processInstanceList, before canceling, can we check that OWNER_TYPE of the candidate instance is ERROR?
 - Can use WorkflowServices.getProcesses() with appropriate query instead of adding new methods to the EngineDataAccess interface.